### DISCUSSION TITLE
Why do some values of categorical features have a low positive rate but get high SHAP scores in binary classification?

### DISCUSSION THREAD

**Discussion by waley22** [shap/shap#3784] [Q&A]
Hi there,

I'm using SHAP for a post hoc attribution analysis for a binary classification problem. To do this, I aggregate the data by each possible bucket value for each feature (for continuous features, I bucketize the feature by quantiles) and check the mean SHAP score. Theoretically, features values that contribute more positively to the result should give a higher mean SHAP score and vice versa; however, I found that for the part of data with some feature values, the positive rate is lower than the overall positive rate, while the SHAP score for this feature is pretty high. 

Is there any theoretical explanation for this phenomenon? 
https://stats.stackexchange.com/questions/535624/are-shap-values-potentially-misleading-when-predictors-are-highly-correlated
This post said that SHAP scores could be potentially misleading when features are correlated. Could that be the main reason?

**Comment by connortann** [comment]
That's an interesting question!

Its hard to say for sure without an example. Do you have a minimal dataset that exhibits this phenomenon?

I think you are right, it might well be related to correlates features. In this case I think it would be helpful to look at a dependence scatter plot, as illustrated here:

https://shap.readthedocs.io/en/latest/example_notebooks/api_examples/plots/scatter.html

Ps, I migrated this issue to a "discussion", as it seems a better fit for here.

**Reply by waley22** [reply]
Thanks for the reply! This is part of my internship project, so the data is confidential and I can't reveal it, and it has ~1M rows and 130 columns with many complex correlations, so I could hardly reproduce it from scratch.

I tried to draw a dependence scatter plot, but it failed due to the presence of categorical features (as stated in this issue: https://github.com/shap/shap/issues/3135 ; seems that it has been fixed recently, but too bad our cluster is still running Python 3.7, so we can only use an older version of SHAP)

One possible explanation might be that some features are too strongly correlated with the label, so these features might overshadow others and mislead the model. This week we tried removing some features that are too obvious, and the mean SHAP score for each bucket seems to align with the positive rate better now. We are also looking for other criterions aside from simply taking the mean SHAP score.
